src/main/java/com/simibubi/create/content/contraptions/fluids/tank/FluidTankBlock.java

```diff
@@ -356,7 +356,55 @@ public boolean hasAnalogOutputSignal(BlockState state) {
 	@Override
 	public int getAnalogOutputSignal(BlockState blockState, Level worldIn, BlockPos pos) {
 		return getTileEntityOptional(worldIn, pos).map(FluidTankTileEntity::getControllerTE)
-			.map(te -> ComparatorUtil.fractionToRedstoneLevel(te.getFillState()))
+			.map(te -> {
+					int output = 0;
+					if (te.isBoiler()) {
+						BoilerData boiler = te.getBoilerData();
+
+						if (boiler.isPassive()) {
+							output = 1;
+						} else {
+							switch (boiler.getBoilerLevel()) {
+							case 0  : break; // boiler is idle
+							case 1  : output = 2;
+								break;
+							case 2  : output = 3;
+								break;
+							case 3  : output = 4;
+								break;
+							case 4  : output = 5;
+								break;
+							case 5  :
+							case 6  : output = 6;
+								break;
+							case 7  :
+							case 8  : output = 7;
+								break;
+							case 9  : output = 8;
+								break;
+							case 10 : output = 9;
+								break;
+							case 11 :
+							case 12 : output = 10;
+								break;
+							case 13 :
+							case 14 : output = 11;
+								break;
+							case 15 : output = 12;
+								break;
+							case 16 : output = 13;
+								break;
+							case 17 : output = 14;
+								break;
+							case 18 : output = 15;
+								break;
+							}
```

Comment:
Maybe use [this](https://stackoverflow.com/questions/17134839/how-does-the-map-function-in-processing-work) function instead?
it returns a float, so just floor or ceil to your liking

Reply:
Thanks for the feedback.

There is actually a gameplay reason for this explicit mapping; it makes it so a user can detect when an boiler is at full power for common boiler sizes. (ie. there's guaranteed to be a detectable difference between passive, level 1, and 2, between level 3, 4, and 5, between 8, 9, 10, and between 17 and 18.)

I can still use the mapping function if you like but this was my reasoning for doing it that way

Reply:
Another option could be to use a static array to represent the mapping instead of a switch statement like:

```java
private static final int[] OUTPUT_SIGNAL_FOR_BOILER_LEVEL = {0, 2, 3, 4, 5, 6, 6, 7, 7, 8, 9, 10, 10, 11, 11, 12, 13, 14, 15};
```